PaintTest/main.cpp

```diff
@@ -0,0 +1,209 @@
+/*****************************************************************************
+ * OpenRCT2, an open source clone of Roller Coaster Tycoon 2.
+ *
+ * OpenRCT2 is the work of many authors, a full list can be found in contributors.md
+ * For more information, visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * A full copy of the GNU General Public License can be found in licence.txt
+ *****************************************************************************/
+#pragma endregion
+
+#include <algorithm>
+#include <string>
+#include <vector>
+
+extern "C" {
+#include "main.h"
+#include "../src/rct2.h"
+#include "../src/ride/ride.h"
+#include "../src/ride/ride_data.h"
+#include "../src/ride/track.h"
+#include "../src/ride/track_data.h"
+}
+
+typedef struct {
+	uint8 rideType;
+	std::vector<uint8> trackTypes;
+} TestCase;
+
+extern const utf8string RideNames[91];
+extern const utf8string TrackNames[256];
+extern const utf8string FlatTrackNames[256];
+
+enum CLIColour {
+	RED,
+	GREEN,
+};
+
+bool gTestColor = true;
+
+static bool CStringEquals(const char *lhs, const char *rhs) {
+	if (lhs == NULL) return rhs == NULL;
+
+	if (rhs == NULL) return false;
+
+	return strcmp(lhs, rhs) == 0;
+}
+
+static bool ShouldUseColor() {
+	if (gTestColor == false) {
+		return false;
+	}
+
+	const char* const term = getenv("TERM");
+	const bool term_supports_color =
+		CStringEquals(term, "xterm") ||
+		CStringEquals(term, "xterm-color") ||
+		CStringEquals(term, "xterm-256color") ||
+		CStringEquals(term, "screen") ||
+		CStringEquals(term, "screen-256color") ||
+		CStringEquals(term, "tmux") ||
+		CStringEquals(term, "tmux-256color") ||
+		CStringEquals(term, "rxvt-unicode") ||
+		CStringEquals(term, "rxvt-unicode-256color") ||
+		CStringEquals(term, "linux") ||
+		CStringEquals(term, "cygwin");
+
+	return term_supports_color;
+}
+
+static const char* GetAnsiColorCode(CLIColour color) {
+	switch (color) {
+		case RED:     return "1";
+		case GREEN:   return "2";
+		default:            return NULL;
+	};
+}
+
+static void ColouredPrintF(CLIColour colour, const char* fmt, ...) {
+	va_list args;
+	va_start(args, fmt);
+
+	if(!ShouldUseColor()) {
+		vprintf(fmt, args);
+		va_end(args);
+		return;
+	}
+
+	printf("\033[0;3%sm", GetAnsiColorCode(colour));
+	vprintf(fmt, args);
+	printf("\033[m");
+	va_end(args);
+}
+
+int main(int argc, char *argv[]) {
+	std::vector<TestCase> testCases;
+
+	for (int i = 0; i < argc; ++i) {
+		char *arg = argv[i];
+		if (strcmp(arg, "--gtest_color=no") == 0) {
+			gTestColor = false;
+		}
+	}
+
+	for (uint8 rideType = 0; rideType < 91; rideType++) {
+		if (!rideIsImplemented(rideType)) {
+			continue;
+		}
+
+		TestCase testCase = {
+				.rideType = rideType
```

Comment:
VS only allows designated initialiser assignments in C files. You will have to change this to 

```
		TestCase testCase = {0};
		testCase.rideType = rideType;
```